DataAccess/Services/UnitOfWork.cs

```diff
@@ -0,0 +1,58 @@
+﻿using DataAccess.Interfaces;
+using DataAccess.Repositories;
+using DomainLibrary.Entities;
+ 
+
+namespace DataAccess.Services
+{
+    public class UnitOfWork : IUnitOfWork
+    {
+        private Context _context;
+        private NoteRepository _noteRepository;
+        private CategoryRepository _categoryRepository;
+        private TemplateRepository _templateRepository;
+        private UserRepository _userRepository;
+
+        public UnitOfWork(Context context)
+        {
+            this._context = context;
+        }
+
+        public IRepository<Note> NoteRepository 
+        {
+            get
+            {
+                return _noteRepository = _noteRepository ?? new NoteRepository(_context);
+            }
+        }
```

Comment:
Since C# 6, you can utilize [expression-bodied members](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/expression-bodied-members#properties). So your code will look like this:
`public IRepository<Note> NoteRepository => _noteRepository ?? (_noteRepository = new NoteRepository (_context));`
It will create a read-only property for you.